crates/autopilot/src/limit_order_quoter.rs

```diff
@@ -131,4 +86,60 @@ impl LimitOrderQuoter {
         Metrics::on_failed(failed_orders.try_into().unwrap());
         Ok(())
     }
+
+    async fn update_surplus_fee(&self, order: &Order) -> Result<()> {
+        let quote = self
+            .quoter
+            .calculate_quote(QuoteParameters {
+                sell_token: order.data.sell_token,
+                buy_token: order.data.buy_token,
+                side: match order.data.kind {
+                    OrderKind::Buy => OrderQuoteSide::Buy {
+                        buy_amount_after_fee: order.data.buy_amount,
+                    },
+                    OrderKind::Sell => OrderQuoteSide::Sell {
+                        sell_amount: SellAmount::BeforeFee {
+                            value: order.data.sell_amount + order.data.fee_amount,
+                        },
+                    },
+                },
+                from: order.metadata.owner,
+                app_data: order.data.app_data,
+                signing_scheme: match &order.signature {
+                    Signature::Eip712(_) => QuoteSigningScheme::Eip712,
+                    Signature::EthSign(_) => QuoteSigningScheme::EthSign,
+                    Signature::Eip1271(signature) => {
+                        // TODO What about SignatureConfiguration like in order_validation.rs?
+                        let additional_gas = self
```

Comment:
I think so, otherwise we might get burned by long-standing ERC-1271 orders (not an issue for market orders).

Reply:
Not sure I understand why we should ever skip validation gas computation in the background task updating the orders. If we are not able to determine the signature validation cost of an order we are probably not able to settle it either so why would we update the `surplus_fee` timestamp which would cause that order to end up in the auction.

Isn't the `eip1271_skip_creation_validation` only relevant for the quoting process while creating the order?

Reply:
Sorry, I think there was some misunderstanding on my part. I understood the `--enable-eip1271-orders` parameter and not the `--eip1271-skip-creation-validation` (which I don't think we need here, since there is no point in including an order if its ERC-1271 order is not valid.

Reply:
Maybe ☝️ makes more sense at a higher level and not the `LimitOrderQuoter` component. The reason I had originally thought about putting it here, is that for market orders, if we accept them in the API, we want to commit to executing. This same "guarantee" is not as strong for limit orders, which are already assumed to be executed when possible (and not right away).

Reply:
👍